bike_rent/tests/test_bike_rent.py

```diff
@@ -1,19 +1,22 @@
-from odoo import exceptions
 from odoo.tests import TransactionCase, tagged
 
 
-@tagged('standard')
+@tagged('bike_rent')
 class TestBikeRent(TransactionCase):
 
     def setUp(self):
         super(TestBikeRent, self).setUp()
+        self.sale_order1 = self.env.ref('bike_rent.website_sale_order_rent1_demo')
+        self.BikeRent = self.env['bike.rent']
 
-    def test_verify_stop_date(self):
-        rent0 = self.env['bike.rent'].create({
-            'bike_id': self.env['product.product'].search([('is_bike', '=', True)], limit=1).id,
-            'partner_id': self.env['res.partner'].search([('name', '=', 'Versada')]).id,
-            'rent_start': '2019-04-30 18:00:00',
-            'rent_stop': '2019-04-20 18:00:00',
-        })
-        with self.assertRaises(exceptions.UserError):
-            rent0._onchange_verify_stop_date()
+    def test_bike_rent_creation_on_action_confirm(self):
+        # Testing if action_confirm creates related bike.rent objects and if object count equals rented products count
+        self.sale_order1.action_confirm()
+        # Should I use filtered here instead?
+        new_bike_rent_records = self.BikeRent.search([('sale_id', '=', self.sale_order1.id)])
+        self.assertTrue(new_bike_rent_records)
+        self.assertEqual(len(new_bike_rent_records), len(self.sale_order1.order_line.product_id.rented_bike_ids))
+
+        # Testing if rent duration data is recorded correctly
+        product_rent_duration = self.sale_order1.order_line.product_id.rent_duration
+        self.assertEqual(product_rent_duration, new_bike_rent_records.read()[0]['rent_time'])
```

Comment:
`.read()[0][field]` is a openerp 7.0 way of getting information, just use `mapped`